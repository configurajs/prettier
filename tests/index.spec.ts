import { describe, expect, it } from 'vitest'
import { defineConfig } from '../dist/index.js'

describe('test config', () => {
  it('option is undefined', () => {
    expect(defineConfig()).toMatchSnapshot()
  })

  it('sortImports is false', () => {
    expect(defineConfig({ sortImports: false })).toMatchSnapshot()
  })

  it('sortAtomicClass is false', () => {
    expect(defineConfig({ sortAtomicClass: false })).toMatchSnapshot()
  })

  it('sortPackageJson is false', () => {
    expect(defineConfig({ sortPackageJson: false })).toMatchSnapshot()
  })

  it('custom importsOrder', () => {
    expect(defineConfig({ importsOrder: ['^@/(.*)$', '^~/(.*)$', '^[.]'] })).toMatchSnapshot()
  })

  it('custom packageJsonOrder', () => {
    expect(defineConfig({ packageJsonOrder: ['name', 'version'] })).toMatchSnapshot()
  })

  it('custom config', () => {
    expect(
      defineConfig({
        semi: true,
        plugins: ['prettier-plugin-test'],
        vueIndentScriptAndStyle: true,
        overrides: [
          {
            files: '*.md',
            options: {
              organizeImportsSkipDestructiveCodeActions: true,
            },
          },
        ],
      }),
    ).toMatchSnapshot()
  })
})
