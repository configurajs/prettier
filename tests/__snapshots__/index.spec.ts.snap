// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test config > custom config 1`] = `
{
  "endOfLine": "auto",
  "importOrder": [
    "<BUILTIN_MODULES>",
    "^vue$",
    "^react$",
    "<THIRD_PARTY_MODULES>",
    "^@/(.*)$",
    "^~/(.*)$",
    "^[.]",
  ],
  "overrides": [
    {
      "files": "*.md",
      "options": {
        "organizeImportsSkipDestructiveCodeActions": true,
      },
    },
  ],
  "packageSortOrder": [],
  "plugins": [
    "@ianvs/prettier-plugin-sort-imports",
    "prettier-plugin-tailwindcss",
    "prettier-plugin-packagejson",
    "prettier-plugin-test",
  ],
  "printWidth": 120,
  "semi": true,
  "singleQuote": true,
  "vueIndentScriptAndStyle": true,
}
`;

exports[`test config > custom importsOrder 1`] = `
{
  "endOfLine": "auto",
  "importOrder": [
    "^@/(.*)$",
    "^~/(.*)$",
    "^[.]",
  ],
  "packageSortOrder": [],
  "plugins": [
    "@ianvs/prettier-plugin-sort-imports",
    "prettier-plugin-tailwindcss",
    "prettier-plugin-packagejson",
  ],
  "printWidth": 120,
  "semi": false,
  "singleQuote": true,
}
`;

exports[`test config > custom packageJsonOrder 1`] = `
{
  "endOfLine": "auto",
  "importOrder": [
    "<BUILTIN_MODULES>",
    "^vue$",
    "^react$",
    "<THIRD_PARTY_MODULES>",
    "^@/(.*)$",
    "^~/(.*)$",
    "^[.]",
  ],
  "packageSortOrder": [
    "name",
    "version",
  ],
  "plugins": [
    "@ianvs/prettier-plugin-sort-imports",
    "prettier-plugin-tailwindcss",
    "prettier-plugin-packagejson",
  ],
  "printWidth": 120,
  "semi": false,
  "singleQuote": true,
}
`;

exports[`test config > option is undefined 1`] = `
{
  "endOfLine": "auto",
  "importOrder": [
    "<BUILTIN_MODULES>",
    "^vue$",
    "^react$",
    "<THIRD_PARTY_MODULES>",
    "^@/(.*)$",
    "^~/(.*)$",
    "^[.]",
  ],
  "packageSortOrder": [],
  "plugins": [
    "@ianvs/prettier-plugin-sort-imports",
    "prettier-plugin-tailwindcss",
    "prettier-plugin-packagejson",
  ],
  "printWidth": 120,
  "semi": false,
  "singleQuote": true,
}
`;

exports[`test config > sortAtomicClass is false 1`] = `
{
  "endOfLine": "auto",
  "importOrder": [
    "<BUILTIN_MODULES>",
    "^vue$",
    "^react$",
    "<THIRD_PARTY_MODULES>",
    "^@/(.*)$",
    "^~/(.*)$",
    "^[.]",
  ],
  "packageSortOrder": [],
  "plugins": [
    "@ianvs/prettier-plugin-sort-imports",
    "prettier-plugin-packagejson",
  ],
  "printWidth": 120,
  "semi": false,
  "singleQuote": true,
}
`;

exports[`test config > sortImports is false 1`] = `
{
  "endOfLine": "auto",
  "packageSortOrder": [],
  "plugins": [
    "prettier-plugin-tailwindcss",
    "prettier-plugin-packagejson",
  ],
  "printWidth": 120,
  "semi": false,
  "singleQuote": true,
}
`;

exports[`test config > sortPackageJson is false 1`] = `
{
  "endOfLine": "auto",
  "importOrder": [
    "<BUILTIN_MODULES>",
    "^vue$",
    "^react$",
    "<THIRD_PARTY_MODULES>",
    "^@/(.*)$",
    "^~/(.*)$",
    "^[.]",
  ],
  "plugins": [
    "@ianvs/prettier-plugin-sort-imports",
    "prettier-plugin-tailwindcss",
  ],
  "printWidth": 120,
  "semi": false,
  "singleQuote": true,
}
`;
